http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/NettyRequestExecutor.java

```diff
@@ -172,6 +173,9 @@ private boolean tryConfigurePipeline() {
         }
 
         pipeline.addLast(LastHttpContentHandler.create());
+        if (Protocol.HTTP2.equals(protocol)) {
+            pipeline.addLast(FlushOnReadHandler.getInstance());
```

Comment:
Do we need to remove `FlushOnReadHandler` from the pipeline when we release the channel?

Reply:
Yep! good catch.